multi_partner_learning.py

```diff
@@ -85,21 +87,21 @@ def compute_test_score_for_single_partner(self, partner):
 
         # Train model
         logger.info("   Training model...")
-        history = model.fit(
-            partner.x_train,
-            partner.y_train,
-            batch_size=partner.batch_size,
-            epochs=self.epoch_count,
-            verbose=0,
-            validation_data=self.val_data,
-            callbacks=cb,
-        )
-
-        # Reference the testset according to the scenario configuration
-        x_test, y_test = self.test_data
+        if isinstance(model, type(LogisticRegression())):
+            history = model.fit(partner.x_train, partner.y_train)
+        else:
+            history = model.fit(
+                partner.x_train,
+                partner.y_train,
+                batch_size=partner.batch_size,
+                epochs=self.epoch_count,
+                verbose=0,
+                validation_data=self.val_data,
+                callbacks=cb,
+            )
 
         # Evaluate trained model
-        model_evaluation = model.evaluate(x_test, y_test, batch_size=constants.DEFAULT_BATCH_SIZE, verbose=0)
+        model_evaluation = self.collaborative_round_evaluation(model, self.test_data)
```

Comment:
We should precise it's evaluated on the test set, so maybe rename to something like `test_evaluation `or `test_score`

Reply:
I ended up using `test_evaluation `and `val_evaluation `to avoid conflicts with the `test_score `attribute in `multi_partner_learning.py` file.